File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-6.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # set a left pointer to the second value
        # this is the tracker pointer as well 

        l = 1
        
        # keep r as the right pointer to find duplicates
        for r in range(1, len(nums)):

            # check if the previous right pointer value is not
            # the same
            if(nums[r-1] != nums[r]):
                # set the left pointer position to the unique right pointer value
                nums[l] = nums[r]

                #increment the left pointer by one
                l += 1

        
        return l 

        

File: Data Structures & Algorithms/duplicate-integer/submission-6.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, bool> nMap;
        for(int n : nums) {
            if(nMap[n]) {
                return true; 
            }

            nMap[n] = true;
        }
        return false;
    }
};